npm/vite-dev-server/src/plugins/cypress.ts

```diff
@@ -105,6 +105,23 @@ export const Cypress = (
         return res.end(transformedIndexHtml)
       })
     },
+    transform (code, id, options?) {
+      try {
+        if (/\.js$/.test(id) && !/\/\/# sourceMappingURL=/i.test(code)) {
+          // The Vite dev server serves esnext, meaning many JS files aren't transpiled thus no sourcemaps are sent along. However,
+          // a sourcemap is generated by Vite which is in effect an "identity" sourcemap mapping 1-to-1 to the output file. We can
+          // grab this and pass it along as a sourcemap we want Vite to embed into the output, giving Cypress a sourcemap to use
+          // for codeFrame lookups.
+
+          return {
+            code,
+            map: this.getCombinedSourcemap(),
```

Comment:
Edit, apparently not - as per your comment above. I still think I'm missing some understanding here, though... if you write something like

```ts
interface Blah {
  foo: string
}

const blah = (): Blah = { 
  throw Error('argh')
  return { foo: "bar" }
}

blah()
```

And serve via Vite - it strips out TS, but this would require a sourcemap to get correct error location - if Vite doesn't generate sourcemaps, how does this error manifest in a regular Vite dev env?

I'll try it out. Hm.

Reply:
Seems Vite is generating a sourcemap normally. I wonder why we need to do this - why can't we do `map: null` and get it like the docs specify? 

![image](https://user-images.githubusercontent.com/19196536/191660513-6ebba93d-c2a9-4e50-807f-78a395283b81.png)

Reply:
Oh right, so you are saying it will automatically append the sourceMap for anything other than plain JS... this is only required for the case of plain JS (eg, does it "just work" for TSX, for example, even if I remove this?) This is my understanding based on your comment. I will try this to find out if my assumption is correct.

Reply:
@lmiller1990 Correct - this is here to handle the case where *source JS* files would not get sourcemaps. Pretty much everything else did without this change

Reply:
Makes sense, since normally you don't need source maps for JS - nothing to map :tada: This worked out surprisingly well, glad Vite exposes the Rollup API.